Crash report 3.2: NPE in StatsActivity.onBlogChanged

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=200, result=-1, data=Intent { (has extras) }} to activity {org.wordpress.android/org.wordpress.android.ui.stats.StatsActivity}: java.lang.NullPointerException
       at android.app.ActivityThread.deliverResults(ActivityThread.java:3433)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:3476)
       at android.app.ActivityThread.access$1200(ActivityThread.java:157)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1337)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:176)
       at android.app.ActivityThread.main(ActivityThread.java:5317)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1102)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:869)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at org.wordpress.android.ui.stats.StatsActivity.onBlogChanged(StatsActivity.java:552)
       at org.wordpress.android.ui.WPActionBarActivity.onActivityResult(WPActionBarActivity.java:604)
       at org.wordpress.android.ui.stats.StatsActivity.onActivityResult(StatsActivity.java:251)
       at android.app.Activity.dispatchActivityResult(Activity.java:5515)
       at android.app.ActivityThread.deliverResults(ActivityThread.java:3429)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:3476)
       at android.app.ActivityThread.access$1200(ActivityThread.java:157)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1337)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:176)
       at android.app.ActivityThread.main(ActivityThread.java:5317)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1102)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:869)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/ViewSiteActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/ViewSiteActivity.java
@@ -53,7 +53,6 @@ public class ViewSiteActivity extends AuthenticatedWebViewActivity {
 
     @Override
     public void onBlogChanged() {
-        super.onBlogChanged();
         mBlog = WordPress.currentBlog;
         loadSiteURL();
     }

--- a/WordPress/src/main/java/org/wordpress/android/ui/WPActionBarActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/WPActionBarActivity.java
@@ -601,7 +601,7 @@ public abstract class WPActionBarActivity extends Activity {
                         setupCurrentBlog();
                     }
                     if (data != null && data.getBooleanExtra(PreferencesActivity.CURRENT_BLOG_CHANGED, true)) {
-                        onBlogChanged();
+                        blogChanged();
                     }
                     WordPress.registerForCloudMessaging(this);
                 }
@@ -628,7 +628,7 @@ public abstract class WPActionBarActivity extends Activity {
             } else {
                 WordPress.setCurrentBlog(blogIDs[position]);
                 updateMenuDrawer();
-                onBlogChanged();
+                blogChanged();
             }
         }
 
@@ -703,7 +703,7 @@ public abstract class WPActionBarActivity extends Activity {
     /**
      * This method is called when the user changes the active blog or hides all blogs
      */
-    public void onBlogChanged() {
+    private void blogChanged() {
         WordPress.wpDB.updateLastBlogId(WordPress.getCurrentLocalTableBlogId());
         // the menu may have changed, we need to change the selection if the selected item
         // is not available in the menu anymore
@@ -729,6 +729,18 @@ public abstract class WPActionBarActivity extends Activity {
         if (shouldUpdateCurrentBlogStatsInBackground()) {
             WordPress.sUpdateCurrentBlogStats.forceRun();
         }
+
+        if (WordPress.getCurrentBlog() != null) {
+            onBlogChanged();
+        }
+
+    }
+
+    /**
+     * This method is called in when the user changes the active blog - descendants should override
+     * this to perform activity-specific updates upon blog change
+     */
+    protected void onBlogChanged() {
     }
 
     /**

--- a/WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsActivity.java
@@ -73,8 +73,6 @@ public class CommentsActivity extends WPActionBarActivity
 
     @Override
     public void onBlogChanged() {
-        super.onBlogChanged();
-
         // clear the backstack
         FragmentManager fm = getFragmentManager();
         fm.popBackStackImmediate(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
@@ -272,7 +272,6 @@ public class MediaBrowserActivity extends WPActionBarActivity implements MediaGr
 
     @Override
     public void onBlogChanged() {
-        super.onBlogChanged();
 
         // clear edit fragment
         if (mMediaEditFragment != null) {

--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java
@@ -589,7 +589,6 @@ public class PostsActivity extends WPActionBarActivity
 
     @Override
     public void onBlogChanged() {
-        super.onBlogChanged();
         popPostDetail();
         attemptToSelectPost();
         mPostList.clear();

--- a/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsActivity.java
@@ -547,15 +547,8 @@ public class StatsActivity extends WPActionBarActivity implements ScrollViewExt.
 
     @Override
     public void onBlogChanged() {
-        super.onBlogChanged();
 
         stopStatsService();
-
-        if (WordPress.getCurrentBlog() == null) {
-            //No blogs, or all blogs are hidden. Close this activity, the app takes you back to the Reader.
-            finish();
-            return;
-        }
 
         mLocalBlogID = WordPress.getCurrentBlog().getLocalTableBlogId();
         scrollToTop();

--- a/WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java
@@ -475,7 +475,6 @@ public class ThemeBrowserActivity extends WPActionBarActivity implements
 
     @Override
     public void onBlogChanged() {
-        super.onBlogChanged();
         if (areThemesAccessible()) {
             fetchThemes(mViewPager.getCurrentItem());
             setRefreshing(true, mViewPager.getCurrentItem());

